packages-exp/auth-exp/test/integration/flows/email.test.ts

```diff
@@ -0,0 +1,148 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect, use } from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+
+import {
+  createUserWithEmailAndPassword,
+  EmailAuthProvider,
+  reload,
+  signInWithCredential,
+  signInWithEmailAndPassword,
+  updateProfile
+} from '@firebase/auth-exp/index.browser';
+import { Auth, OperationType, UserCredential } from '@firebase/auth-types-exp';
+import { FirebaseError } from '@firebase/util';
+
+import {
+  cleanUpTestInstance,
+  getTestInstance,
+  randomEmail
+} from '../../helpers/integration/helpers';
+
+use(chaiAsPromised);
+
+describe('Integration test: email/password auth', () => {
+  let auth: Auth;
+  let email: string;
+  beforeEach(() => {
+    auth = getTestInstance();
+    email = randomEmail();
+  });
+
+  afterEach(() => cleanUpTestInstance(auth));
+
+  it('allows user to sign up', async () => {
+    const userCred = await createUserWithEmailAndPassword(
+      auth,
+      email,
+      'password'
+    );
+    expect(auth.currentUser).to.eq(userCred.user);
+    expect(userCred.operationType).to.eq(OperationType.SIGN_IN);
+
+    const user = userCred.user;
+    expect(user.isAnonymous).to.be.false;
+    expect(user.uid).to.be.a('string');
+    expect(user.email).to.eq(email);
+    expect(user.emailVerified).to.be.false;
+  });
+
+  context('with existing user', () => {
+    let signUpCred: UserCredential;
+
+    beforeEach(async () => {
+      signUpCred = await createUserWithEmailAndPassword(
```

Comment:
Perhaps a test where createUserWithEmailAndPassword is called twice in a row with the same details?

Reply:
Done